lib/database.js

```diff
@@ -189,7 +189,8 @@ class ItemDatabase {
 	 * @param {string} name - The name of the item to add.
 	 * @param {number} count - The count of item to remove.
 	 */
-	addItem(name, count) {
+	addItem(name, countInput) {
+		var count = Number(countInput);
```

Comment:
This is not right.  addItem is supposed to throw an error if you pass it something that's not a number.  Have you got a stack trace for when a non-number was passed to it?

Reply:
> This is not right. addItem is supposed to throw an error if you pass it something that's not a number. Have you got a stack trace for when a non-number was passed to it?

It simply happend every time I put something in the sender object in Factorio.

Then the fix should be here instead:
Within /api/place (app.post), req.body.count is a string at this point.
db.items.addItem(req.body.name, req.body.count);

Reply:
It is the correct way of doing type conversion at the calling side, not inside the function. If anything, function should check if the incoming parameter is a number.

So yeah, please update the addItem call, not the addItem function itself.